src/open_inwoner/accounts/signals.py

```diff
@@ -95,27 +95,31 @@ def _update_eherkenning_user_from_kvk_api(user: User):
 
         # Update company name. We want this to run on every login because it may
         # change (if infrequently)
-        if vestiging := (
-            kvk_client.get_vestiging(vestiging=user.vestiging)
-            if user.vestiging
-            else kvk_client.get_company_headquarters(kvk=user.kvk)
-        ):
-            if (
-                company_name := vestiging.get("naam")
-            ) and user.company_name != company_name:
-                user.company_name = company_name
-                user.save(update_fields=["company_name"])
-                updated_fields.append("company_name")
+        basisprofiel = kvk_client.get_basisprofiel(kvk=user.kvk)
+        company_name = basisprofiel.get("naam")
+        if company_name and user.company_name != company_name:
+            user.company_name = company_name
+            user.save(update_fields=["company_name"])
+            updated_fields.append("company_name")
+
+        # Update branch name (use hoofdvestiging for rechtspersoon)
```

Comment:
We don't appear to use the hoofdvestiging anymore: that's fine I think, because the hoofdvestiging is still a vestiging and would be its own user, so the rechtspersoon should be something different (as here the first handelsnaam, but see comment below)

```suggestion
        # Update branch name (use first handelsnaam for rechtspersoon)
```